src/panels/lovelace/elements/hui-state-label-element.ts

```diff
@@ -0,0 +1,78 @@
+import { html, LitElement } from "@polymer/lit-element";
+
+import "../../../components/entity/ha-state-label-badge.js";
+
+import { computeTooltip } from "../../../common/string/compute-tooltip";
+import computeStateDisplay from "../../../common/entity/compute_state_display.js";
+import { handleClick } from "../../../common/dom/handle-click";
+import { longPress } from "../common/directives/long-press-directive";
+import { hassLocalizeLitMixin } from "../../../mixins/lit-localize-mixin";
+import { LovelaceElement, LovelaceElementConfig } from "./types.js";
+import { HomeAssistant } from "../../../types.js";
+import { TemplateResult } from "lit-html";
+
+interface Config extends LovelaceElementConfig {
+  prefix?: string;
+  suffix?: string;
+}
+
+class HuiStateLabelElement extends hassLocalizeLitMixin(LitElement)
+  implements LovelaceElement {
+  public hass?: HomeAssistant;
+  private _config?: Config;
+
+  static get properties() {
+    return { hass: {}, _config: {} };
+  }
+
+  public setConfig(config: Config): void {
+    if (!config.entity) {
+      throw Error("Invalid Configuration: 'entity' required");
+    }
+
+    this._config = config;
+  }
+
+  protected render(): TemplateResult {
+    if (!this._config) {
+      return html``;
+    }
+
+    const state = this.hass!.states[this._config.entity!];
+    return html`
+      ${this.renderStyle()}
+      <div
+        .title="${computeTooltip(this.hass!, this._config)}"
+        @ha-click="${() => handleClick(this, this.hass!, this._config!, false)}"
+        @ha-hold="${() => handleClick(this, this.hass!, this._config!, true)}"
```

Comment:
Do you need these `!` on the above 3 lines for hass and config?

Reply:
Yes, according to the linter.

Reply:
OHH
I see. You  are checking for `if(this.config)` but not hass. So you shouldnt need it for config atleast. You might as well check if hass is undefined there as well and take the `!` out

Reply:
I have done so in previous PRs, but @balloob has told me a few times that it wasn't necessary. I'll see if I can find the comment

Reply:
Gotcha. Well at least you can take it out of the configs

Reply:
I can take what out of the configs?

Reply:
The `!`. Should not be needed as you check if it exists

`!` denotes you are saying it exists for sure even though it could be undefined. but here it wont ever be undefined because of the if statements

Reply:
This is of the this._config!

Reply:
One would think, but the linter complains when I remove those for the event handlers